### PR TITLE
Add the REF-by-consumer authoring-shape sweep

### PR DESCRIPTION
**Pull request by hhenson**
**Stack 2 of N**, based on #648 (architecture ratchets). Retarget to `main` once #648 merges.

Adds `python/tests/test_ref_consumer_sweep.py`: 765 products of

- **8 input shapes**: `TS[int]`, `TS[CompoundScalar]` (plain and with derived leaves), `TS[tuple]`, `TSD[str, ...]`, `TSD[CompoundKey, ...]` (polymorphic key, the #521 shape), `TSS`, fixed `TSL`, `TSB`
- **8 REF-producing sources**: a `REF`-typed node, fixed `TSL` projection, `TSD` item lookup, `map_` element, `switch_` branch, a switch that flips from a value body to a REF body, `if_(...).true`, `default(nothing(...), ts)`
- **~60 consumers**: every std operator that accepts the shape, field access, `combine`, `collect`, `mesh_`, `dispatch`, a Python compute node

Oracle: the REF arm's `eval_node` trace must equal the plain arm's. The plain arm is asserted separately so a consumer-definition mistake cannot masquerade as a runtime defect. No 0.5 oracle is needed, so it also covers C++-first-only shapes. Runs in under two seconds.

**What it found on `main` today** (recorded as `xfail(strict=True)`, 90 products):
- #649 `reduce` and `mesh_` reject a REF-valued `TSD`/`TSL` (wiring error and a runtime ops-kind failure) for five of the eight sources; a `TSL` projection and a `switch_` branch do *not* reproduce it, so the from-REF adaptation is applied on some binding paths and not others (family 1 of the retrospective).
- #650 a `switch_` whose active branch flips from a value body to a REF body never ticks again: silent data loss, every shape, every consumer except the REF-transparent ones (family 2).

Design record: `docs/source/developer_guide/testing.rst`, "Authoring-shape sweeps".

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01EsqG5G3RzpnPtLCJ62DFga

### CODE REVIEW

File: python/tests/test_ref_consumer_sweep.py [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+        "drop": lambda ts: drop(ts, 1),
+        "dedup": lambda ts: dedup(ts),
+        "default": lambda ts: default(nothing(TS[int]), ts),
+        "filter_": lambda ts: filter_(const(True), ts),
+        "if_then_else": lambda ts: if_then_else(const(True), ts, ts),
+        "merge": lambda ts: merge(ts, ts),
+        "race": lambda ts: race(ts, nothing(TS[int])),
+        "gate": lambda ts: gate(const(True), ts),
+        "sum_": lambda ts: sum_(ts),
+        "count": lambda ts: count(ts),
```
[the remark this answers is not in the document]
Comment (hhenson) [reply python/tests/test_ref_consumer_sweep.py]: Fixed in 9fff7d034. The downstream hgraph_analytics count import and consumer were removed, so the core wheel test extra is self-contained.

File: python/tests/test_ref_consumer_sweep.py [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+binding inserts the from-REF adaptation, and a type variable binds the
+dereferenced schema. The 2026-09-04 fix-series retrospective found that rule
+patched at ten consumers (collect, mesh keys, CompoundScalar fields, ungroup,
+tuple combine, nested bundle projections, ...), which is why this sweep
+exists.
+
+Oracle: for every (shape, source, consumer) product, the graph that feeds the
+consumer through a REF-producing source yields the same ``eval_node`` trace
+as the graph that feeds it the plain source. No released-hgraph oracle is
+needed, so the sweep also covers C++-first-only shapes.
```
[the remark this answers is not in the document]
Comment (hhenson) [reply python/tests/test_ref_consumer_sweep.py]: Addressed. The sweep documentation now points to existing native public-wiring coverage in test_graph_wiring.cpp for scalar and structural REF round trips, plus test_ref_executor.cpp for dynamic retargeting through a value consumer. The stack also adds dedicated native switch REF transition tests in PR #655, avoiding a duplicate C++ cartesian copy of the Python authoring sweep.

File: python/tests/test_ref_consumer_sweep.py [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+def test_plain_arm_evaluates(shape_id, consumer_id):
+    """The oracle arm must itself run: a failure here is a consumer definition bug."""
+    trace = _expected(shape_id, consumer_id)
+    assert isinstance(trace, list) and trace
+
+
+@pytest.mark.parametrize(
+    "case",
+    [
+        pytest.param(case, marks=pytest.mark.xfail(reason=_known_gap(case), strict=True))
```
[the remark this answers is not in the document]
Comment (hhenson) [reply python/tests/test_ref_consumer_sweep.py]: Fixed in 9fff7d034. Known gaps now declare an exact exception type and message or expected invalidated-source trace, which is asserted before xfail. Any other exception or trace mismatch fails normally. The later fixes in this stack empty the gap tables.

File: python/tests/test_ref_consumer_sweep.py [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+    "TSD[str, TS[int]]": Shape(
+        TSD[str, TS[int]], [{"a": 1, "b": 2}, {"a": 3}, None, {"b": REMOVE, "c": 5}]
+    ),
+    "TSD[Key, TS[int]]": Shape(
+        TSD[Key, TS[int]],
+        [
+            {Key("one", "a"): 1, Key("two", "b"): 2},
+            {Key("one", "a"): 3},
+            None,
+            {Key("two", "b"): REMOVE, Key("three", "c"): 5},
```
[the remark this answers is not in the document]
Comment (hhenson) [reply python/tests/test_ref_consumer_sweep.py]: Fixed in 9fff7d034. The polymorphic-key input now includes a SweepKey and SweepDerivedKey together in the first tick and removes both later, exercising mixed realized key types across the REF path.
